contentlayer.config.ts

```diff
@@ -1,18 +1,116 @@
-export const _DataName = defineDocumentType(() => ({
-  name: "Post", // name of the post
-  filePathPattern: `**/*.md`, // file path where the mdx file to be process is located
-  contentType: "mdx", // content type mdx
-  // fields: types desc
+import * as fs from "fs";
+import { defineDocumentType, defineNestedType, makeSource } from "contentlayer2/source-files";
+
+const Resources = defineNestedType(() => ({
+  name: "Resources",
   fields: {
-    title: { type: "string", required: true },
-    date: { type: "date", required: true },
-  },
-  computedFields: {
-    url: { type: "string", resolve: (post) => `/posts/${post._raw.flattenedPath}` },
+    title: { type: "string" },
+    url: { type: "string" },
   },
 }));
 
-export default makeSource({ contentDirPath: "posts", documentTypes: [_DataName] });
+const path = `${process.cwd()}/public/bitcoin-transcript`;
+
+const generateDocuTypes = () => {
+  const getFolders = fs.readdirSync(path, "utf-8");
+
+  // getFolders is sliced @6 to remove files which are not folders like .gitignore, readme.md , .github etc. We only want to read files from the folders that contain transcripts.
+  const folders = getFolders.slice(6).filter((item) => item !== "twitter_handles.json");
+
+  // firstLetterRegex: ** /(^\w{1})|(\s+\w{1})/g ** gets the first letter of the split folder name and capitalizes the first letter of the each word. This is to create a camelCase naming Convention for the generated folders
+  const firstLetterRegex = /(^\w{1})|(\s+\w{1})/g;
+
+  const DefinedDocumentTypes = folders.map((name) => {
+    const slugifyName = name
+      .split("-")
+      .map((text) => text.replace(firstLetterRegex, (text) => text.toUpperCase()))
+      .join("");
+
+    return defineDocumentType(() => ({
+      name: slugifyName,
+      filePathPattern: `${name}/**/*.md`,
+      contentType: "markdown",
+      fields: {
+        title: { type: "string" },
+        speakers: { type: "list", of: { type: "string" } },
+        date: { type: "date" },
+        transcript_by: { type: "string" },
+        Transcript_by: { type: "string" },
+        "transcript by": { type: "string" },
+        categories: { type: "list", of: { type: "string" } },
+        tag: { type: "list", of: { type: "string" } },
+        tags: { type: "list", of: { type: "string" } },
+        media: { type: "string" },
+        translation_by: { type: "string" },
+        episdoe: { type: "number" },
+        episode: { type: "number" },
+        aliases: { type: "list", of: { type: "string" } },
+        vídeo: { type: "string" },
+        video: { type: "string" },
+        hosts: { type: "list", of: { type: "string" } },
+        source: { type: "string" },
+        transcription_coverage: { type: "string" },
+        summary: { type: "string" },
+        needs: { type: "string" },
+        aditional_resources: { type: "list", of: Resources },
+        additional_resources: { type: "list", of: Resources },
```

Comment:
Does the content layer have some utility that tells you that some documents have this misspelled field? How did you come up with the need for `aditional_resources` and `additional_resources`?

Reply:
Yes it has an additional utility that tells you fields that are not added from the md files

Reply:
How do I replicate that?